From ec4dbc686a4840e3680d812269c7a8f5ac97c245 Mon Sep 17 00:00:00 2001
From: Kuiying Wang <kuiying.wang@intel.com>
Date: Tue, 23 Oct 2018 11:55:14 +0800
Subject: [PATCH] Add power control interface

methods:
    setPowerState
    getPowerState
    forcePowerOff
properties:
    PGood
    State

Note:
All the interfaces are  the low level interface to handle HW.
And provides supporting the high level interface like state-manager.
They are the common interfaces and all the platforms implement them all.

Change-Id: I6d90439879f5a4940cad03fd588dd233f81f2802
Signed-off-by: Kuiying Wang <kuiying.wang@intel.com>
---
 .../Chassis/Control/Power.interface.yaml           | 61 ++++++++++++++++++++++
 1 file changed, 61 insertions(+)
 create mode 100644 xyz/openbmc_project/Chassis/Control/Power.interface.yaml

diff --git a/xyz/openbmc_project/Chassis/Control/Power.interface.yaml b/xyz/openbmc_project/Chassis/Control/Power.interface.yaml
new file mode 100644
index 0000000..740a1c0
--- /dev/null
+++ b/xyz/openbmc_project/Chassis/Control/Power.interface.yaml
@@ -0,0 +1,61 @@
+description: >
+    Power control service
+methods:
+    - name: setPowerState
+      description: >
+        set host power state.
+      parameters:
+        - name: state
+          type: int32
+          description: >
+            0 for force power off host
+            1 for power on host
+      returns:
+        - name: status
+          type: int32
+          description: >
+            0: means success
+            -1: means other errors not throwing out.
+      errors:
+       - xyz.openbmc_project.Chassis.Common.Error.UnsupportedCommand
+       - xyz.openbmc_project.Chassis.Common.Error.IOError
+
+    - name: getPowerState
+      description: >
+        Get current host power status.
+      returns:
+        - name: status
+          type: int32
+          description: >
+            Current host status,
+            0 for host power off
+            1 for host power on
+      errors:
+       - xyz.openbmc_project.Chassis.Common.Error.UnsupportedCommand
+       - xyz.openbmc_project.Chassis.Common.Error.IOError
+
+    - name: forcePowerOff
+      description: >
+        Force power off the host.
+      returns:
+        - name: status
+          type: int32
+          description: >
+            The result of power off command.
+      errors:
+       - xyz.openbmc_project.Chassis.Common.Error.UnsupportedCommand
+       - xyz.openbmc_project.Chassis.Common.Error.IOError
+
+properties:
+    - name: PGood
+      type: int32
+      default: 0
+      description: >
+       PSU Power good property
+    - name: State
+      type: int32
+      default: 0
+      description: >
+       System power status,
+       0: power is off
+       1: power is on
\ No newline at end of file
-- 
2.7.4

